pkg/cloudprovider/cloudapi/azure/azure_nsg_rules.go

```diff
@@ -556,20 +593,40 @@ func convertFromAzureSecurityRuleToInternalEgressRule(rule armnetwork.SecurityRu
 		return nil, err
 	}
 
+	// special handling for Azure any rule, where both dstIP and securityGroups will be empty.
+	if len(dstIP) == 0 && len(securityGroups) == 0 {
```

Comment:
This is for user created rules. You need all the rules to be present in the sync content right.

Reply:
Yes.